### PR TITLE
feat: add params for initContainers, and hostNetwork

### PR DESCRIPTION
**Pull request by bodji**

### CODE REVIEW

File: operations/pyroscope/helm/pyroscope/templates/services.yaml [hunk 1]
```diff
@@ -9,6 +10,10 @@ metadata:
   labels:
     {{- include "pyroscope.labels" . | nindent 4 }}
     app.kubernetes.io/component: {{ $component | quote }}
+  annotations:
+  {{- with $values.serviceAnnotations }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
```
Comment (kolesnikovae) [inline operations/pyroscope/helm/pyroscope/templates/services.yaml:16]: There is an open PR https://github.com/grafana/pyroscope/pull/2364/files: we should maybe go that way. What do you think?

**Review by kolesnikovae — COMMENTED** [review]
Thank you for the contribution! 🎉 

The change looks good. I've left a couple of comments. Also, may I ask you to sign the CLA please?

File: operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml [hunk 2]
```diff
@@ -154,7 +161,7 @@ spec:
   {{- end }}
 {{- if eq $values.kind "StatefulSet" }}
 ---
-apiVersion: policy/v1
+apiVersion: policy/v1beta1
```
[the remark this answers is not in the document]
Comment (bodji) [reply operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:164]: I had to change this to make it work on our v1.20.15

File: operations/pyroscope/helm/pyroscope/templates/services.yaml (hunk 1 above)
Comment (bodji) [reply operations/pyroscope/helm/pyroscope/templates/services.yaml:16]: Yes ok, I'll remove it from this one

File: operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kolesnikovae) [reply operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:164]: I see, I'm sorry about that. We should probably make DisruptionBudget optional or set the api version in a smarter way:

```yaml
{{- if .Capabilities.APIVersions.Has "policy/v1" }}
apiVersion: policy/v1
{{- else }}
apiVersion: policy/v1beta1
{{- end }}
```

@bodji would you like to create another PR for this? :)

**Comment by bodji** [comment]
I signed the CLA.

I removed serviceAnnotations (already brought by: https://github.com/grafana/pyroscope/pull/2364)

I removed the v1beta for pod disruption budget

File: operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml [hunk 3]
```diff
@@ -41,6 +41,13 @@ spec:
       serviceAccountName: {{ include "pyroscope.serviceAccountName" . }}
       securityContext:
         {{- toYaml $values.podSecurityContext | nindent 8 }}
+      {{- if hasKey $values "hostNetwork" }}
+      hostNetwork: {{ $values.hostNetwork }}
+      {{- end }}
+      initContainers:
+      {{- with $values.initContainers }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
```
Comment (kolesnikovae) [inline operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:50]: I think we should declare an empty collection in `values.yaml`: 
```yaml
initContainers: []
```

**Review by kolesnikovae — COMMENTED** [review]
Could you also please run `make helm/check` and `make helm/docs` to update rendered manifests and the documentation

File: operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml (hunk 3 above)
Comment (kolesnikovae) [inline operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:50]:
```suggestion
      {{- with $values.initContainers }}
      initContainers:
      {{- toYaml . | nindent 8 }}
      {{- end }}
```

File: operations/pyroscope/helm/pyroscope/rendered/micro-services.yaml [hunk 4]
```diff
@@ -56,7 +56,7 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: "minio-sa"
-  namespace: "default"
+  namespace: "pyroscope"
```
Comment (kolesnikovae) [inline operations/pyroscope/helm/pyroscope/rendered/micro-services.yaml:59]: I think we should use the `default` namespace by default

Comment (bodji) [reply operations/pyroscope/helm/pyroscope/rendered/micro-services.yaml:59]: Where is this specified ? It gets changed when I do make helm/check

**Comment by cyriltovena** [comment]
> I added a "-n default" to the helm/check helm commands in the makefile.
> 
> Because by default, it takes the current namespace set in the kube context
> 
> What do you think ?

perfect !

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**